docs/LanguageGuide.md

```diff
@@ -496,3 +496,22 @@ Here is a little overview of the API:
 ; if you want to replace every occurrence, use -1
 (Pattern.substitute #"sub-me" "sub-me sub-me sub-me" "replaced" -1) ; => "replaced replaced replaced"
 ```
+
+#### Limitations of Patterns
+
+As mentioned above, patterns are not as expressive as Regular Expressions. The
+fundamental difference is that patterns do not backtrack. This means that they
+cannot express alternation (because we can’t go back to where we branched) and
+we cannot reduce non-greedy matches on the left. The latter point might not be
+obvious, so let us look at an example:
+
+```
+(Pattern.match-all-groups #"1.-2" "1 1 2") ; => [[@"1 1 2"]]
+```
+
+A valid, less greedy match would have been `"1 2"`, but since it would have
+required us to go back to the left after we had started matching to reduce the
+match size, this is not done. As such, while `-` is similar to `*?`, it is not
```

Comment:
Great, that certainly helps!

I'd add 
```
is similar to `*?` from RegularExpressions
```
to point out, where this syntax originated and where it can (not) be used.

Reply:
Thanks for the feedback, done!